### PR TITLE
fall back to service catalogue url if fail with MultipleChoice error after api version discovery

### PR DESCRIPTION
**Pull request by hmlanigan**
Fix lp:1817242.

If SendRequest() fails with MultipleChoice error after api version discovery, fall back to service catalogue url.  Previously fall back was for all services.  Now it's on a per service basis.

### CODE REVIEW

**Comment by hmlanigan** [comment]
This change can be tested against the openstack cloud in the bug:
https://bugs.launchpad.net/juju/+bug/1817242

Here are juju cloud and credential setup for my trial account:
https://pastebin.canonical.com/p/ccQChHYj9c/

novarc for openstack trial account:
https://pastebin.canonical.com/p/MsDhYVkvFh/

Create image metadata to bootstrap with:
source novarc
mkdir ~/simplestreams
juju metadata generate-image -d ~/simplestreams  -r $OS_REGION_NAME -u $OS_AUTH_URL -s xenial -i f99bb6f7-658d-4f7c-840d-40228ee22581
juju metadata generate-image -d ~/simplestreams  -r $OS_REGION_NAME -u $OS_AUTH_URL -s bionic -i 02e47341-017f-4cc5-a049-92bbeeb7310f

juju bootstrap  bluvalt --metadata-source ~/simplestreams
juju deploy ubuntu

File: client/apiversion.go [hunk 1]
```diff
@@ -13,39 +13,39 @@ import (
 	"gopkg.in/goose.v2/logging"
 )
 
-type apiVersion struct {
-	major int
-	minor int
+type ApiVersion struct {
```
Comment (howbazaar) [inline client/apiversion.go:19]: Public types should have a doc comment.

**Review by howbazaar — APPROVED** [review]
Looking good.

File: client/apiversion.go (hunk 1 above)
Comment (hmlanigan) [reply client/apiversion.go:19]: added comments.

**Pull request merged**